File: UserDemo/src/main/java/com/nakul/user/controller/UserController.java
package com.nakul.user.controller;

import com.nakul.user.model.User;
import com.nakul.user.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api")
public class UserController {

    @Autowired
    private UserService userService;

    @GetMapping("/user")
    public List<User> get() {
        return userService.get();
    }

    @GetMapping("/user/{id}")
    public User get(@PathVariable int id) {
        return userService.get(id);
    }

    @PutMapping("/user")
    public User update(@RequestBody User user) {
        return userService.save(user);
    }

    @PostMapping("/user")
    public User save(@RequestBody User user) {
        return userService.save(user);
    }

    @DeleteMapping("/user/{id}")
    public String delete(@PathVariable int id) {
        userService.delete(id);
        return "Employee Deleted";
    }
}

File: UserKotlin/src/main/kotlin/com/nakul/user/controller/UserController.kt
package com.nakul.user.controller

import com.nakul.user.dto.request.CreateUserRequestDTO
import com.nakul.user.dto.request.LoginRequestDTO
import com.nakul.user.dto.response.UserResponseDTO
import com.nakul.user.service.UserService
import jakarta.validation.Valid
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.transaction.annotation.Transactional
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api")
class UserController {

    @Autowired
    private lateinit var userService: UserService

    @GetMapping("/user")
    fun get(): List<UserResponseDTO> = userService.read()

    @GetMapping("/user/{id}")
    fun get(@PathVariable("id") userId: Int): UserResponseDTO = userService.read(userId = userId)

    @PutMapping("/user/{id}")
    @Transactional
    fun update(@PathVariable("id") userId: Int, @RequestBody createUserDTO: CreateUserRequestDTO): UserResponseDTO =
        userService.update(userId = userId, createUserDTO = createUserDTO)


    @DeleteMapping("/user/{id}")
    @Transactional
    fun delete(@PathVariable id: Int): UserResponseDTO = userService.delete(id)


    @PostMapping("/login")
    @Transactional
    fun login(@RequestBody loginDTO: LoginRequestDTO): UserResponseDTO? {
        return userService.login(loginDTO = loginDTO)
    }

    @PostMapping("/signUp")
    @Transactional
    fun signUp(@Valid @RequestBody createUserDTO: CreateUserRequestDTO): UserResponseDTO {
        return userService.create(createUserDTO)
    }

}